libpod/runtime_img.go

```diff
@@ -276,24 +277,51 @@ func DownloadFromFile(reader *os.File) (string, error) {
 
 // LoadImage loads a container image into local storage
 func (r *Runtime) LoadImage(ctx context.Context, inputFile string, writer io.Writer, signaturePolicy string) (string, error) {
-	if newImages, err := r.LoadAllImageFromArchive(ctx, writer, inputFile, signaturePolicy); err == nil {
+	newImages, err := r.LoadAllImageFromArchive(ctx, writer, inputFile, signaturePolicy)
+	if err == nil {
 		return newImages, nil
 	}
+	logrus.Warnf("Error trying pulling multi-arch images: %v", err)
 	return r.LoadImageFromSingleImageArchive(ctx, writer, inputFile, signaturePolicy)
 }
 
 // LoadAllImageFromArchive loads all images from the archive of multi-image that inputFile points to.
 func (r *Runtime) LoadAllImageFromArchive(ctx context.Context, writer io.Writer, inputFile, signaturePolicy string) (string, error) {
-	newImages, err := r.ImageRuntime().LoadAllImagesFromDockerArchive(ctx, inputFile, signaturePolicy, writer)
+	var latestErr error
+	appendErr := func(e error) {
+		if latestErr != nil {
+			latestErr = errors.Wrapf(latestErr, "%v\n", e)
+		} else {
+			latestErr = e
+		}
+	}
+	newImageNames, err := r.ImageRuntime().LoadAllImagesFromDockerArchive(ctx, inputFile, signaturePolicy, writer)
+	if err == nil {
+		return strings.Join(newImageNames, ", "), nil
+	}
+	appendErr(errors.Wrapf(err, "error trying multi docker aechive"))
+	newImageNames, err = r.ImageRuntime().LoadAllImagesFromOCIArchive(ctx, inputFile, signaturePolicy, writer)
 	if err == nil {
-		return getImageNames(newImages), nil
+		return strings.Join(newImageNames, ", "), nil
 	}
-	return "", err
+	appendErr(errors.Wrapf(err, "error trying multi oci aechive"))
+	return "", latestErr
 }
 
 // LoadImageFromSingleImageArchive load image from the archive of single image that inputFile points to.
 func (r *Runtime) LoadImageFromSingleImageArchive(ctx context.Context, writer io.Writer, inputFile, signaturePolicy string) (string, error) {
-	var err error
+	var (
+		src       types.ImageReference
+		err       error
+		latestErr error
+	)
+	appendErr := func(e error) {
+		if latestErr != nil {
+			latestErr = errors.Wrapf(latestErr, "error pulling images: %v\n", e)
```

Comment:
This will prefix some errors but not all? If I am reading this correctly,
```
error pulling images: error pulling images: error1
error2
error3
```
---

`s/latestErr/allErrors/` maybe?

---

Most importantly, compare `pkg/errorhandling`, or maybe use it. I don’t know what are all of the coding conventions in Podman, but this does look like a problem appropriate for a common solution.